src/graph.rs

```diff
@@ -988,4 +1149,145 @@ mod tests {
+        // center is 7
+        let (s0, s1, s2, s3, s4, s5, s6, s7) = ("0", "1", "2", "3", "4", "5", "6", "7");
+        let addresses = vec!["0", "1", "2", "3", "4", "5", "6", "7"];
+        let mut graph: Graph<&str> = Graph::new();
+        graph.insert(Edge::new(s0, s7));
+        graph.insert(Edge::new(s1, s7));
+        graph.insert(Edge::new(s2, s7));
+        graph.insert(Edge::new(s3, s7));
+        graph.insert(Edge::new(s4, s7));
+        graph.insert(Edge::new(s5, s7));
+        graph.insert(Edge::new(s6, s7));
+
+        let agraph = graph.create_agraph(&addresses);
+        let (betweenness, closeness) = graph.compute_betweenness_and_closeness(&agraph);
+
+        let total_path_length = [13, 13, 13, 13, 13, 13, 13, 7];
+        let num_paths = [7, 7, 7, 7, 7, 7, 7, 7];
+        let mut expected_closeness: [f64; 8] = [0.0; 8];
+        for i in 0..8 {
+            expected_closeness[i] = total_path_length[i] as f64 / num_paths[i] as f64;
+        }
+        assert_eq!(betweenness, [0, 0, 0, 0, 0, 0, 0, 21]);
+        assert_eq!(closeness, expected_closeness);
+    }
+
+    #[test]
+    fn disconnected_graph() {
+        // 9 vertices
+        // 4 verts, 0-3: square, all points connected
+        // 5 verts, 4-8: star, with v4 in the center
+        let (s0, s1, s2, s3, s4, s5, s6, s7, s8) = ("0", "1", "2", "3", "4", "5", "6", "7", "8");
+        let addresses = vec!["0", "1", "2", "3", "4", "5", "6", "7", "8"];
+        let mut graph: Graph<&str> = Graph::new();
+        graph.insert(Edge::new(s0, s1));
+        graph.insert(Edge::new(s0, s2));
+        graph.insert(Edge::new(s0, s3));
+        graph.insert(Edge::new(s1, s2));
+        graph.insert(Edge::new(s1, s3));
+        graph.insert(Edge::new(s2, s3));
+
+        graph.insert(Edge::new(s4, s5));
+        graph.insert(Edge::new(s4, s6));
+        graph.insert(Edge::new(s4, s7));
+        graph.insert(Edge::new(s4, s8));
+
+        let agraph = graph.create_agraph(&addresses);
+        let (betweenness, closeness) = graph.compute_betweenness_and_closeness(&agraph);
+
+        let total_path_length = [3, 3, 3, 3, 4, 7, 7, 7, 7];
+        let num_paths = [3, 3, 3, 3, 4, 4, 4, 4, 4];
+        let mut expected_closeness: [f64; 9] = [0.0; 9];
+        for i in 0..9 {
+            expected_closeness[i] = total_path_length[i] as f64 / num_paths[i] as f64;
+        }
+        assert_eq!(betweenness, [0, 0, 0, 0, 6, 0, 0, 0, 0]);
+        assert_eq!(closeness, expected_closeness);
+    }
+
+    // Helper function to create an AGraph from a json file.
+    // The file will begin like this:
+    //   {"agraph":[[2630,3217,1608,1035,...
+    // and end like this:
+    //   ...2316,1068,1238,704,2013]]}
+    pub fn load_agraph(filepath: &str) -> AGraph {
+        let jstring = fs::read_to_string(filepath).unwrap();
+        let agraph_sample: AGraphSample = serde_json::from_str(&jstring).unwrap();
+        let agraph = agraph_sample.agraph;
+        agraph
+    }
+
+    #[test]
```

Comment:
I'd ignore this case by defaut, since it's is time-consuming to run. 
```suggestion
    #[test]
    #[ignore = "takes a while to run"]
```

Reply:
It might be better suited as a bench in future, no need to change that now though.

Reply:
done